### ISSUE TITLE
Customize serialization of query parameter lists

### ISSUE THREAD

**Issue by puremath** [emberjs/data#7345]
# Background

If we specify a list as query parameters in the `query` method of the `RESTAdapter` (or with the JSONAPIAdapter as it inherits the `query` method), such as

```
this.store.query('book', {"filter: {"authors": [1,2]}})
```
we have a situation, which is not specified in the standard defined on jsonapi.org.

# Situation in Ember Data (RESTAdapter and JSONAPIAdapter)

To serialize the query parameters we use a function, which was copied from jQuery and therefore get the same result as with jquery:

```
book?filter[authors][]=1&filter[authors][]=2
```

Source: https://github.com/emberjs/data/blob/v3.21.0/packages/adapter/addon/-private/utils/serialize-query-params.ts#L11

Since it is all happening in private methods, this behaviour cannot be easily changed for a backend which behaves differently - such as Django Rest Framework Json API.

# Situation in Django DJA

Django Rest Framework JSON Api (the Django implementation of jsonapi.org) decided to implement query parameter lists as repeated query parameter like this:

```
book?filter[authors]=1&filter[authors]=2
```
Source: https://github.com/django-json-api/django-rest-framework-json-api/issues/718

# Solutions

There should definitely be a way to easily change this behaviour to adapt it to different back ends, especially since it is not defined in the standard.

I'm not sure how this can be resolved. Maybe the function `serializeQueryParams(queryParamsObject)` could be made a public method on the RESTAdapter.

**Comment by cah-brian-gantzler** [comment]
Bump, just ran into this need. While mine isnt nested, I get the following
```
this.store.query('book', {"authors": [1,2]})
book?authors[]=1&authors[]=2
```
and spring java wants
```
book?authors=1&authors=2
```

**Comment by runspired** [comment]
@cah-brian-gantzler it's unlikely we ever consider this. Those adapters have too many poorly designed hooks and methods as it is. It is much simpler to implement your own adapter.

Eventually we will provide url-building util functions, even in that case the answer wouldn't be a hook but to write your own function.

**Comment by cah-brian-gantzler** [comment]
I believe when I looked into this the functions that performed this was so way down or in utils that could not be overridden. If I could have overridden a method on the base adapter I would have.  Im sorry I dont remember where I saw the code needed to be changed, but I dont think writing my own adapter would solve the problem either, I think it was in the build url utilities. 

Any idea on the time frame of eventually 😄 

Thanks for your consideration
